src/datumaro/experimental/converters/annotation_converters.py

```diff
@@ -506,3 +511,859 @@ def rotate_corner(expr: pl.Expr):
+
+    def convert(self, df: pl.DataFrame) -> pl.DataFrame:
+        """Convert ellipse data to target dtype."""
+        return _convert_fixed_array_dtype(
+            df,
+            self.input_ellipse.name,
+            self.output_ellipse.name,
+            self.output_ellipse.field.dtype,
+            array_size=4,  # Ellipse has 4 elements like bbox
+        )
+
+
+@converter
+class EllipseCoordinateConverter(Converter):
+    """
+    Convert ellipse coordinates between normalized and absolute formats.
+
+    This converter handles transformations between normalized coordinates
+    (range [0,1]) and absolute pixel coordinates using image dimensions.
+    """
+
+    input_ellipse: AttributeSpec[EllipseField]
+    input_image: AttributeSpec[ImageField]
+    output_ellipse: AttributeSpec[EllipseField]
+
+    def filter_output_spec(self) -> bool:
+        """
+        Check if ellipse normalization conversion is needed and configure output.
+
+        Returns:
+            True if conversion is needed (normalization status differs), False otherwise
+        """
+        input_normalized = self.input_ellipse.field.normalize
+        output_normalized = self.output_ellipse.field.normalize
+
+        # Configure output specification with correct normalization
+        self.output_ellipse = AttributeSpec(
+            name=self.output_ellipse.name,
+            field=EllipseField(
+                semantic=self.input_ellipse.field.semantic,
+                dtype=self.input_ellipse.field.dtype,
+                format=self.input_ellipse.field.format,
+                normalize=output_normalized,
+            ),
+        )
+
+        # Apply converter only if normalization status needs to change
+        return input_normalized != output_normalized
+
+    def convert(self, df: pl.DataFrame) -> pl.DataFrame:
+        """
+        Convert ellipse coordinates between normalized and absolute formats.
+
+        Uses image dimensions to transform coordinates. For normalized to absolute:
+        multiplies by image dimensions. For absolute to normalized: divides by
+        image dimensions.
+
+        Args:
+            df: Input DataFrame containing ellipse and image data
+
+        Returns:
+            DataFrame with converted ellipse coordinates
+        """
+        input_ellipse_name = self.input_ellipse.name
+        input_image_name = self.input_image.name
+        output_ellipse_name = self.output_ellipse.name
+
+        # Get the image shape column name
+        image_shape_name = f"{input_image_name}_shape"
+
+        input_normalized = self.input_ellipse.field.normalize
+
+        # Coordinate order for width/height mapping: [width, height, width, height] for x1y1x2y2
+        coordinates_order = [1, 0, 1, 0]
+
+        def op(x: pl.Expr, y: pl.Expr) -> pl.Expr:
+            """Choose operation based on conversion direction."""
+            xy = x * y if input_normalized else x.cast(pl.Float64) / y
```

Comment:
At L184 you removed the workaround to force-cast to Float64; can't you also remove it here and the other coordinate converters?